src/tir/analysis/check_contains.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file check_contains.h
+ * \brief Interface of the analysis that tells if an expression contains
+           a node that satisfies a given predicate.
+ */
+
+#ifndef TVM_TIR_ANALYSIS_CHECK_CONTAINS_H_
+#define TVM_TIR_ANALYSIS_CHECK_CONTAINS_H_
+
+#include <tvm/tir/expr.h>
+#include <tvm/tir/stmt_functor.h>  // For the class StmtExprVisitor
+
+namespace tvm {
+namespace tir {
+
+/*!
+ * \brief Visitor which tells if a given expression or statement contains a subexpression
+          that satisfies a given predicate
+ */
+class CheckContains : public StmtExprVisitor {
```

Comment:
This class looks very similar to https://github.com/apache/tvm/blob/5012462ef88acfd6a84b3f28135b361a8788f257/src/tir/analysis/var_touch.cc#L30

Is it possible to merge them? e.g. Refactor `VarTouch` as a special case of `CheckContains`.

Reply:
I'll see if VarTouch is a special case of my new analyzer CheckContains.
But if that's the case, I would prefer to do the appropriate changes to VarTouch (or the occurrences that use it), into another PR, in order to let things clear and minimal.